### ISSUE TITLE
Support configuring GCFS / image streaming

### ISSUE THREAD

**Issue by lauraseidler** [terraform-google-modules/terraform-google-kubernetes-engine#1149]
### TL;DR

Support configuring GCFS / enabling image streaming by introducing a new variable that sets the required parameter in the node configuration.

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#gcfs_config
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#node_config
```

### Detailed design

```markdown
Image streaming can be enabled for clusters and/or node pools by adding a block

gcfs_config {
  enabled = true/false
}
```
to the node config. 

I propose to add a variable `enable_image_streaming`/`enable_gcfs` or similar to allow setting this block. I would default it to `false` as that's the current behaviour of the module.

Potentially, setting this variable to `true` could also automatically enable the `containerfilesystem.googleapis.com` API, though this might be out of scope.
```


### Additional information

Currently, there's no way to enable image streaming on a node pool that's managed by this module, as the absence of this block will force the node pool to be re-created without it even if image streaming is enabled manually.

Adding this block always as either true/false might also solve some other issues related to this block, e.g. https://github.com/hashicorp/terraform-provider-google/issues/11017.

If this is an acceptable feature request, I'm happy to provide a PR :)

**Comment by morgante** [comment]
Happy to review a PR adding this.
